### PR TITLE
Task-57165: News mosaic template is ko if small width

### PR DESCRIPTION
**Pull request by hchaabene**
Prior to this change, first publish 4 articles with names not truncated  , when i move latest new template in a small container or using mobile device to the right side or in a column , then  choose mosaic model , 
This is due to changing the width is small, the display is bad and names of articles are truncated.
After this change,  the article is readable , and names of articles are not truncated.

### CODE REVIEW

File: webapp/src/main/webapp/news-list-view/components/views/NewsMosaicView.vue [hunk 1]
```diff
@@ -76,20 +87,24 @@ export default {
...
+      isSmallWidth: false
     };
   },
   created() {
     this.reset();
     this.$root.$on('saved-news-settings', this.refreshNewsViews);
     this.getNewsList();
   },
+  updated() {
+    this.isSmallWidth =  this.$refs.top?.clientWidth *100 / window.screen.width  < 33;
```
Comment (ahamdi) [inline webapp/src/main/webapp/news-list-view/components/views/NewsMosaicView.vue]: Why are you using this check ? couldn't we use Mobile vs Tablet vs Dersktop?

Comment (hchaabene) [reply webapp/src/main/webapp/news-list-view/components/views/NewsMosaicView.vue]: this method allows to detect the dimension of the screen and to check if the display of the mosaic template is by column or by the normal form.

**Review by ahamdi — DISMISSED** [review]

File: webapp/src/main/webapp/news-list-view/components/views/NewsMosaicView.vue [hunk 2]
```diff
@@ -15,12 +15,12 @@ You should have received a copy of the GNU Affero General Public License
 along with this program. If not, see <http://www.gnu.org/licenses/>.
 -->
 <template>
-  <div id="top-news-mosaic">
+  <div id="top-news-mosaic" ref="top">
```
Comment (ahamdi) [inline webapp/src/main/webapp/news-list-view/components/views/NewsMosaicView.vue]:
```suggestion
  <div id="top-news-mosaic" ref="top-news-mosaic">
```

File: webapp/src/main/webapp/news-list-view/components/views/NewsMosaicView.vue [hunk 3]
```diff
@@ -76,20 +76,24 @@ export default {
...
+      isSmallWidth: false
     };
   },
   created() {
     this.reset();
     this.$root.$on('saved-news-settings', this.refreshNewsViews);
     this.getNewsList();
   },
+  mounted() {
+    this.isSmallWidth =  this.$refs.top?.clientWidth *100 / window.screen.width  < 33;
```
Comment (ahamdi) [inline webapp/src/main/webapp/news-list-view/components/views/NewsMosaicView.vue]: Could you adjust the reference ID here too ?

File: webapp/src/main/webapp/news-list-view/components/views/NewsMosaicView.vue [hunk 4]
```diff
@@ -102,6 +106,9 @@ export default {
           .finally(() => this.initialized = false);
       }
     },
+    minLength(lengthNews){
+      return lengthNews<5 && lengthNews>0 ? 100/lengthNews : 25;
```
Comment (ahamdi) [inline webapp/src/main/webapp/news-list-view/components/views/NewsMosaicView.vue]:
```suggestion
      return lengthNews < 5 && lengthNews > 0 ? 100 / lengthNews : 25;
```

**Review by ahamdi — APPROVED** [review]

**Pull request merged**